index.html

```diff
@@ -3288,95 +3262,6 @@ <h3><dfn>Writing or pushing content</dfn></h3>
       </p>
       </section>
 
-      <section><h3>Mapping JSON to NDEF</h3>
-      <p>
-        To <dfn>map JSON to NDEF</dfn> given a |record:NDEFRecordInit|,
-        run these steps:
-        <ol class=algorithm>
-          <li>
-            Let |mimeTypeRecord| be the <a>MIME type</a>
-            returned by running <a>parse a MIME type</a> on |record|'s mediaType.
-            <ol>
-              <li>
-                If |mimeTypeRecord| is failure, let |mimeTypeRecord| be a new
-                <a>MIME type record</a> whose type is "`application`", and subtype is "`json`".
-              </li>
-              <li>
-                If |mimeTypeRecord| is not a <a>JSON MIME type</a>,
-                then [= exception/throw =] a
-                {{"SyntaxError"}} {{DOMException}} and abort these steps.
-              </li>
-            </ol>
-          </li>
-          <li>
-            Let |data:byte sequence| be an empty [= byte sequence =].
-            <ol>
-              <li>
-                Let |stream:byte stream| be the resulting
-                <a data-cite="encoding#concept-stream">byte stream</a> of executing
-                <a>serialize JSON to bytes</a> on |record|'s data.
-              </li>
-              <li>
-                <a data-cite="encoding#concept-stream-read">Read</a> bytes from |stream| into
-                |data| until <a data-cite="encoding#concept-stream-read">read</a> returns
-                <a data-cite="encoding#end-of-stream">end-of-stream</a>.
-              </li>
-            </ol>
-          </li>
-          <li>
-            Set |length:unsigned long| to the [=byte sequence/length=]
-            of |data|.
-          </li>
-          <li>
-            Let |ndefRecord| be the notation for the <a>NDEF record</a> to
-            be created by the UA.
-            <ol>
-              <li>
-                Set the |ndefRecord|'s <a>TNF field</a> to `2` (<a>MIME type</a>).
-              </li>
-              <li>
-                Set the |ndefRecord|'s <a>TYPE field</a> to the result of
-                <a>serialize a MIME type</a> with |mimeTypeRecord| as
-                the input.
-              </li>
-              <li>
-                Set the |ndefRecord|'s <a>PAYLOAD LENGTH field</a> to |length|.
-              </li>
-              <li>
-                If |length| > `0`, set the |ndefRecord|'s <a>PAYLOAD field</a> to |data|.
-              </li>
-            </ol>
-          </li>
-          <li>
-            Return |ndefRecord|.
-          </li>
-        </ol>
-      </p>
-      <aside class=note>
-        <p>
-          JavaScript Object Notation aka JSON can be any of the following [[[ECMASCRIPT]]]
-          types: object, array, String, Boolean, Number or null, which covers a lot of
-          the [[[WEBIDL]]] types as listed here: [= JSON type =].
-        </p>
-        <p>
-          Serializing JSON to bytes can result in 0 bytes in some cases, like serializing
```

Comment:
I still think we might want such a note about payload might become empty somewhere

Reply:
If the data is empty, it makes sense that payload is empty right?
```js
const writer = new NFCWriter();
writer.push({records: [
    {
      recordType: "media",
      mediaType: "application/json",
      data: new ArrayBuffer()
    }
]});
// OR
writer.push(new ArrayBuffer());
```

Reply:
Yes for creation, but it may not be a valid NDEF record in all cases, in which case the script might get an error later (when trying to push the record and the underlying platform rejecting it).

Reply:
I'm not seeing this restriction on Android: https://android.googlesource.com/platform/frameworks/base/+/master/core/java/android/nfc/NdefRecord.java#413
Does spec forbids empty payload?

Reply:
Right, the spec should not make a restriction here, either. If something is not right, the underlying platform will tell. In this case, Android won't seem to object :)

Reply:
The NDEF spec actually allows this (I don't know from where do I have the memory of being invalid in some cases).

> A payload length of 0 is allowed in which case the PAYLOAD field is omitted from the NDEF
record

Reply:
Actually that's the only normative prose I found.

> If the PAYLOAD_LENGTH field value is 0, the PAYLOAD field MUST NOT be present.

Reply:
But a developer might be confused that writing JSON can actually result in reading back an empty payload - that is the note that I would like

Reply:
I'm not sure which JSON would you write that result in an empty payload?

Reply:
The example showed how :)

Reply:
![image](https://user-images.githubusercontent.com/1101339/67010517-13685c80-f0ee-11e9-916e-ee683a2905cb.png)

Reply:
Having a `toJSON` method on a class seems very unlikely and not related to Web NFC.